### PR TITLE
ISSUE: 3066. CORS support for /version

### PR DESCRIPTION
**Pull request by McMutton**
Adds CORS support for /version

### CODE REVIEW

File: src/app/contextBroker/orionRestServices.cpp [hunk 1]
```diff
@@ -506,9 +507,8 @@ static RestService badVerbV[] =
 
 /* ****************************************************************************
 *
-* optionsV - options service vector to be used if CORS is enabled
+* optionsV - 
 *
-* Adds API_V2_CORS definitions on top of the default service vector (restServiceV)
 */
```
Comment (kzangeli) [inline src/app/contextBroker/orionRestServices.cpp]: Not sure why the comment about "options service vector to be used if CORS is enabled" was removed.
It's valid, right?

File: src/app/contextBroker/orionRestServices.cpp [hunk 2]
```diff
@@ -551,6 +552,7 @@ void orionRestServicesInit
    const char*         httpsCert
 )
 {
+  //Use options service vector (optionsServiceV) only when CORS is enabled
```
Comment (kzangeli) [inline src/app/contextBroker/orionRestServices.cpp]: Ah ... the comment was moved here ... ?
If so ... All OK (I guess) , but the Orion style guide 'demands' `//` to be followed by a space

File: src/lib/serviceRoutines/optionsVersionRequest.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  ConnectionInfo*            ciP,
+  int                        components,
+  std::vector<std::string>&  compV,
+  ParseData*                 parseDataP
+)
+{
+  if (isOriginAllowedForCORS(ciP->httpHeaders.origin))
+  {
+    ciP->httpHeader.push_back(ACCESS_CONTROL_ALLOW_ORIGIN);
+    // If any origin is allowed, the header is sent always with "any" as value
```
Comment (kzangeli) [inline src/lib/serviceRoutines/optionsVersionRequest.cpp]: Adding a newline whenever "needed" makes the code easier to read ...

File: src/lib/serviceRoutines/versionTreat.cpp [hunk 4]
```diff
@@ -79,6 +81,23 @@ std::string versionTreat
   ParseData*                 parseDataP
 )
 {
+  if (isOriginAllowedForCORS(ciP->httpHeaders.origin))
+  {
+    ciP->httpHeader.push_back(ACCESS_CONTROL_ALLOW_ORIGIN);
+    // If any origin is allowed, the header is sent always with "any" as value
```
Comment (kzangeli) [inline src/lib/serviceRoutines/versionTreat.cpp]: Comment says "any", code seems to say "*" ...
If so, please add ("*") to comment to make things clear

**Comment by fgalan** [comment]
Nothing to add to what @kzangeli has said. Thus, I'm giving LGTM and passing the ball to @kzangeli 

Btw, good to see a new contribution from you @McMutton ! It has passed a long time :)

File: src/app/contextBroker/orionRestServices.cpp (hunk 2 above)
Comment (McMutton) [reply src/app/contextBroker/orionRestServices.cpp]: Yes I moved the comment here. It seemed to make more sense to have the comment where the decision is made.

Fixed in: b87cdd8

File: src/app/contextBroker/orionRestServices.cpp (hunk 1 above)
Comment (McMutton) [reply src/app/contextBroker/orionRestServices.cpp]: Please see my comment below. NTC

File: src/lib/serviceRoutines/optionsVersionRequest.cpp (hunk 3 above)
Comment (McMutton) [reply src/lib/serviceRoutines/optionsVersionRequest.cpp]: Fixed in: b87cdd8

File: src/lib/serviceRoutines/versionTreat.cpp (hunk 4 above)
Comment (McMutton) [reply src/lib/serviceRoutines/versionTreat.cpp]: Fixed in: b87cdd8

**Comment by McMutton** [comment]
> Btw, good to see a new contribution from you @McMutton ! It has passed a long time :)

Thanks @fgalan, it is good to be back :) Although I would have liked to be more active, it was not possible due to a very busy period.

**Comment by fgalan** [comment]
Upon merging this PR, CI platform has reported this problem:

```
-----  CORS version request  -----
(0501_cors/version_request.test) output not as expected
VALIDATION ERROR: input line:
   Content-Length: 370
does not match ref line:
   Content-Length: 357
```
A similar case happened recently with a PR authored by @saurabhjangir 

Easty to fix (just an one-line PR changint that line, I'll manage that) but a bit weird...

**Comment by fgalan** [comment]
I have realized what was happening... GET /version use to have a variable payload, so we always use `REGEX(\+d)` for Conent-Length in that case. Have a look to fixing PR at https://github.com/telefonicaid/fiware-orion/pull/3143

**Comment by McMutton** [comment]
Ok I got it, I'll keep this in mind.

Just out of curiosity, why do we have a variable payload?

**Comment by fgalan** [comment]
Because some of the fields of the version response are variable (e.g. uptime, compilation date, compilation host, etc.).

**Comment by McMutton** [comment]
Right.. This is so obvious. Sorry for the stupid question :)

**Pull request merged**